app/views/pages/report.ejs

```diff
@@ -555,4 +737,276 @@
   var obfuscatePath = true;
 </script>
 
+<script>
+  /* Page validation functions. 
+  If validation fails and messages to be displayed should be dispatched in the validation function.
+  These aren't done */
+  const validateDemographicsAndGeography = () => {
+      //validate age
+      var hasAge = document.getElementById("Age").value != "";
+      if (!hasAge) {
+        return [false,"<%= __('Please select a age range') %>"];
+      }
+
+      //validate sex
+      var hasGender = radioSet(document.getElementsByName("gender"));
+      if (!hasGender) {
+        return [false,"<%= __('Please select a Biological gender/Sex') %>"];
+      }
+
+      //validate zip/postal code
+      const isInt = parseInt(document.getElementById("postal-code").value);
```

Comment:
Postal code contains letters for many countries. We already have a pattern on the input element, so we can just let the browser handle the validation by using `document.getElementById("postal-code").checkValidity()`. This can possibly also be used for other validations

Reply:
I've just tried using .checkValidity(). It returns true for a 3 character zip code when a four digit one is required by the pattern. This test was conducted in firefox. I've removed the is int check so now postal code validation just checks for the right length the input.